lib/elm-test.js

```diff
@@ -258,21 +258,16 @@ function resolveGlobs(fileGlobs) {
   return _.flatMap(globs, resolveFilePath);
 }
 
-function getWatchPaths(projectRootDir /*:string*/) {
-  var watchedSourcePaths;
-  var elmJson = fs.readJsonSync(path.join(projectRootDir, 'elm.json'), 'utf8');
+function getWatchPaths(elmJson) {
+  let watchedSourcePaths;
   if (elmJson['type'] === 'package') {
-    watchedSourcePaths = ['./src'];
+    watchedSourcePaths = ['src'];
   } else {
     watchedSourcePaths = elmJson['source-directories'];
   }
-  var watchedTestPaths = path.join(projectRootDir, 'tests');
-  var watchedPaths = watchedSourcePaths
-    .concat(watchedTestPaths)
-    .map(function(sourcePath) {
-      return path.resolve(projectRootDir, sourcePath) + '/**/*.elm';
-    });
-  return watchedPaths;
+  return [...watchedSourcePaths, 'tests'].map(function(sourcePath) {
+    return path.posix.join(sourcePath, '**', '*.elm');
```

Comment:
What happens if I run this from a directory that isn't the project root?

Before, it would join with the project root to make that not matter. Now it appears to work relative to the current directory, wherever that might be.

This seems like a potential regression!

Reply:
Thanks for the review! In [line 452](https://github.com/rtfeldman/node-test-runner/pull/349/files/ba14ad1b33696edb8907a1a4e5f2d0c4b8e0c3f6#diff-047f238be7300455dd55a8c22225e53fR452) I set chokidar's working directory to `projectRootDir`.

Therefore, chokidar works relative to _its_ current directory which is the _project's_ root directory. Therefore, I believe this PR does not introduce a regression.

Saying that, it is clear that the intention of the code is not clear - I believe I can rename a variable or two to clarify which paths are relative to what.

Reply:
@rtfeldman I went ahead and checked if running elm-test from a subdirectory of a project with my changes and found #355.

> What happens if I run this from a directory that isn't the project root?

Currently elm-test crashes due to a different bug!